src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleFormatterOptions.cs

```diff
@@ -32,22 +32,6 @@ public class ConsoleFormatterOptions
         /// </summary>
         public bool UseUtcTimestamp { get; set; }
 
-        internal virtual void Configure(IConfiguration configuration)
-        {
-            if (ConsoleLoggerOptions.ParseBool(configuration, nameof(IncludeScopes), out bool includeScopes))
-            {
-                IncludeScopes = includeScopes;
-            }
```

Comment:
The new code genn'd code looks like:

```C#
            foreach (IConfigurationSection section in configuration.GetChildren())
            {
                switch (section.Key)
                {
                    case "IncludeScopes":
                        {
                            if (configuration["IncludeScopes"] is string stringValue21)
                            {
                                obj.IncludeScopes = ParseBool(stringValue21, () => section.Path)!;
                            }
                        }
                        break;
```

Why do we need a `foreach` through all the children, a switch statement, and then the indexer for the specific property? That is too much overhead (For example, GetChildren always allocs - a lot - because it uses LINQ). I would think the code genn'd code should look like this hand-written code.

Reply:
It was written this way to honor `binderOptions.ErrorOnUnknownConfiguration`. We could optimize the code by doing checking that option first, and have a fast path when it's false. 

```cs
if (binderOptions?.ErrorOnUnknownConfiguation is not true)
{
    BindIncludeScopes(configuration["IncludeScopes"]);
    BindUseUtcTimestamp(configuration["UseUtcTimestamp"]);
    ...
}
else
{
    foreach (IConfigurationSection section in configuration.GetChildren())
    {
        switch (section.Key)
        {
            case "IncludeScopes":
                BindIncludeScopes(section.Value);
                break;
            ...
        }
    }
}

void BindIncludeScopes(string sectionValue)
{
    if (sectionValue is not null)
    {
        obj.IncludeScopes = ParseBool(stringValue21, () => section.Path)!; 
    } 
}

...
```

Think that's worth it given the code bloat? Or would you lean toward not using the generator here?

Reply:
As we discussed offline, I'm peeling this off - https://github.com/dotnet/runtime/issues/88110.